File: app/src/main/java/com/lskycity/androidtools/app/ToolApplication.java
package com.lskycity.androidtools.app;

import android.app.Application;

import com.android.volley.RequestQueue;
import com.android.volley.toolbox.Volley;

/**
 *  main application
 *
 */
public class ToolApplication extends Application {
    public static ToolApplication instance;

    private RequestQueue mRequestQueue;

    public ToolApplication() {
        instance = this;
    }

    public static ToolApplication get() {
        return  instance;
    }

    public RequestQueue getRequestQueue() {
        if(mRequestQueue==null) {
            mRequestQueue = Volley.newRequestQueue(this);
        }
        return mRequestQueue;
    }
}

File: app/src/main/java/com/lskycity/androidtools/app/ToolApplication.kt
package com.lskycity.androidtools.app

import android.app.Application
import com.android.volley.RequestQueue
import com.android.volley.toolbox.Volley

/**
 * main application
 *
 */
class ToolApplication : Application() {

    val requestQueue: RequestQueue by lazy { Volley.newRequestQueue(this) }

    init {
        instance = this
    }

    companion object {
        private lateinit var instance: ToolApplication

        fun get(): ToolApplication {
            return instance
        }
    }
}
